### PR TITLE
Remove the trappable_error_type

### PR DESCRIPTION
**Pull request by elliottt**
The `trappable_error_type` argument to `wasmtime::bindgen!` became optional in wasmtime-21, and given that we never expect to trap in any of our host implementations, let's remove that additional complexity.

The meat of this change is removing the custom `FastlyError` type, and replacing it with `types::Error` which is generated from compute.wit. This simplifies some functions like the `v2` suffixed request functions, but in most cases ended up being a renaming from `FastlyError` to `types::Error`.

### CODE REVIEW

File: lib/src/component/http_req.rs [hunk 1]
```diff
@@ -396,32 +398,26 @@ impl http_req::Host for Session {
         h: http_types::RequestHandle,
         b: http_types::BodyHandle,
         backend_name: String,
-    ) -> Result<
-        // This return type is a little surprising. Please see the [error-detail] note at the top
-        // of the file for an explanation.
```
Comment (acw) [inline lib/src/component/http_req.rs:401]: yeah, glad to see this go

File: lib/wit/deps/fastly/compute.wit [hunk 2]
```diff
@@ -136,10 +136,10 @@ interface uap {
...
 
   resource user-agent {
-    family: func(max-len: u64) -> string;
-    major: func(max-len: u64) -> string;
-    minor: func(max-len: u64) -> string;
-    patch: func(max-len: u64) -> string;
+    family: func(max-len: u64) -> result<string, error>;
+    major: func(max-len: u64) -> result<string, error>;
+    minor: func(max-len: u64) -> result<string, error>;
+    patch: func(max-len: u64) -> result<string, error>;
```
Comment (acw) [inline lib/wit/deps/fastly/compute.wit:142]: This bit seems unrelated; is this an intentional inclusion in this PR?

File: lib/src/component/http_req.rs [hunk 3]
```diff
@@ -396,32 +398,26 @@ impl http_req::Host for Session {
...
-    > {
+    ) -> Result<http_types::Response, (Option<http_req::SendErrorDetail>, types::Error)> {
         // This initial implementation ignores the error detail field
-        match self.send(h, b, backend_name).await {
-            Ok(r) => Ok(Ok(r)),
-            Err(e) => e.with_empty_detail(),
-        }
+        self.send(h, b, backend_name)
+            .await
+            .map_err(types::Error::with_empty_detail)
```
Comment (acw) [inline lib/src/component/http_req.rs:405]: If all errors here are going to return `(None, e)`, why are we including the `None`? Is this a future expansion thing?

**Review by acw — CHANGES_REQUESTED** [review]
Just a couple questions!

File: lib/wit/deps/fastly/compute.wit (hunk 2 above)
Comment (elliottt) [reply lib/wit/deps/fastly/compute.wit:142]: Yes. This was an omission from the original pr: those functions pass in a max-length for the resulting string, but had no way to signal an error. Given that I was making these changes, I updated this at the same time.

File: lib/src/component/http_req.rs (hunk 3 above)
Comment (elliottt) [reply lib/src/component/http_req.rs:405]: Yep. The wiggle implementation always sets the error-detail to a default value, but in the wit version, I opted to represent that as an `option<send-error-detail>` instead. Returning `None` is the same as the wiggle version, but I didn't want to break our ability to return the additional detail later.

**Review by acw — APPROVED** [review]
Works for me!
